build_defs.bzl

```diff
@@ -21,13 +21,19 @@ def _c_name(name):
         return name + "_c"
     return name
 
+def _GetPath(ctx, path):
+    if ctx.label.workspace_root:
+        return ctx.label.workspace_root + "/" + path
+    else:
+        return path
+
 def _includes(ctx):
     includes = []
     for include in ctx.attr.includes:
         if include == ".":
-            includes.append(ctx.build_file_path[:ctx.build_file_path.rfind("/")])
+            includes.append(_GetPath(ctx, ctx.label.package))
```

Comment:
Could you share your case why you need this change? If I see the [docs](https://bazel.build/rules/lib/builtins/ctx#build_file_path), apparently `ctx.build_file_path` is deprecated and `ctx.label.package` is recommended.